File: 2021/Day-02/two.py
with open('input.txt', 'r') as f:
    depth = 0
    pos = 0
    aim = 0
    aim_depth = 0

    for line in f:
        line = line.rstrip()
        distance = int(line.split(' ')[1])
        if line[0] == 'f':
            pos += distance
            aim_depth += (aim * distance)
        elif line[0] == 'd':
            # part one
            depth += distance
            # part two
            aim += distance
        else:
            depth -= distance
            aim -= distance

print(depth * pos)
print(aim_depth * pos)

File: 2021/Day-02/two.js
// I also want to use the early, easy puzzles to practice javascript
// I'm running this in node, not a browser window

// Load in the FileSystem module to read files
const { dir } = require('console');
var fs = require('fs');

let input = fs.readFileSync('input.txt', 'utf8');

input = input.split('\n');

// part one

let pos = 0;
let depth = 0;
let aim = 0;
let aim_depth = 0;

for (let i = 0; i < input.length; i++) {
    let line = input[i].split(' ');
    if (line[0] == '') {
        // it seems that js reads in the terminal newline
        continue;
    }
    let direction = line[0];
    let dist = parseInt(line[1]);
    
    switch (direction.charAt(0)) {
        case 'f':
            pos += dist;
            aim_depth += dist * aim;
            break;
        case 'd':
            depth += dist;
            aim += dist;
            break;
        case 'u':
            depth -= dist;
            aim -= dist;
            break;
    }
    
}

console.log(pos * depth);
console.log(pos * aim_depth);